db-tests/src/test/java/gov/nasa/jpl/aerie/database/PlanCollaborationTests.java

```diff
@@ -594,24 +595,26 @@ void snapshotInheritsAllLatestAsParents() throws SQLException{
 
       //take n snapshots, then insert them all into the latest table
       final int numberOfSnapshots = 4;
-      final int[] snapshotIds = new int[numberOfSnapshots];
-      for(int i = 0; i < numberOfSnapshots; ++i){
-        snapshotIds[i] = createSnapshot(planId);
+      final var snapshotIds = new ArrayList<Integer>(numberOfSnapshots);
+      for(int i = 0; i < numberOfSnapshots; i++){
+        snapshotIds.add(createSnapshot(planId));
       }
 
       try(final var statement = connection.createStatement()) {
-        //assert that there is exactly one entry for this plan in plan_latest_snapshot
-        var res = statement.executeQuery(
-            """
-            select snapshot_id from plan_latest_snapshot where plan_id = %d;
-            """.formatted(planId));
-        assertTrue(res.first());
-        assertTrue(res.isLast());
+        {
+          //assert that there is exactly one entry for this plan in plan_latest_snapshot
+          final var res = statement.executeQuery(
+              """
+                  select count(1) from plan_latest_snapshot where plan_id = %d;
+                  """.formatted(planId));
+          res.next();
+          assertEquals(1, res.getInt(1));
+        }
```

Comment:
This change removes the assertion described in the comment. The old code checked that there was exactly one row. The new code checks that the first row's first column equals 1.

Reply:
Good catch - I can add an `assertFalse(res.next())` - or read the resultset into a list and check that its length is one

Reply:
Done